core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -461,7 +461,15 @@ trait SubmissionComponent {
       implicit val getWorkflowMessagesListResult = GetResult { r =>
         val workflowRec = WorkflowRecord(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
         val rootEntityTypeOption: Option[String] = r.<<
-        val entityRec = workflowRec.workflowEntityId.map(EntityRecord(_, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<))
+
+        // val entityRec = workflowRec.workflowEntityId.map(EntityRecord(_, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<))
+        val entityRec = workflowRec.workflowEntityId match {
+          case Some(eid) => Option(EntityRecord(eid, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<))
+          case None =>
+            // consume the entity record columns, but ignore the result
+            val _ = EntityRecord(-1, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
+            None
+        }
```

Comment:
you could also solve this by re-ordering the column selection in WorkflowAndMessagesRawSqlQuery.action so the messages come before the entity, so it doesn't matter if you consume the entity columns or not